src/Compiler/Driver/CompilerImports.fs

```diff
@@ -88,18 +119,23 @@ let MakeILResource rName bytes =
         MetadataIndex = NoMetadataIdx
     }
 
-let PickleToResource inMem file (g: TcGlobals) scope rName p x =
+let PickleToResource inMem file (g: TcGlobals) compress scope rName p x =
     let file = PathMap.apply g.pathMap file
 
-    let bytes = pickleObjWithDanglingCcus inMem file g scope p x
+    let bytes =
+        let bytes = pickleObjWithDanglingCcus inMem file g scope p x
 
-    let byteStorage =
-        if inMem then
-            ByteStorage.FromMemoryAndCopy(bytes.AsMemory(), useBackingMemoryMappedFile = true)
+        if compress then
+            let raw = new MemoryStream(bytes.AsMemory().ToArray())
+            let compressed = new MemoryStream()
+            use deflator = new DeflateStream(compressed, CompressionLevel.Optimal)
+            raw.CopyTo deflator
+            deflator.Close()
+            compressed.ToArray()
         else
-            ByteStorage.FromByteArray(bytes.AsMemory().ToArray())
+            bytes.AsMemory().ToArray()
 
-    (bytes :> IDisposable).Dispose()
```

Comment:
This is not being disposed anymore, is that ok?

Reply:
That is an interesting question.  It looks like memory and I have never been a fan of disposing of memory, I guess I should take a look and see what it actually holds.  Good eyes and thanks for the question.

Reply:
it's now just an array of bytes.  Which doesn't support IDisposable.

Reply:
Well the new wrapper around it is just an array, but the original `bytes: ByteBuffer` is still inside of it and not being disposed. And it seems to be using the array pool to which it's supposed to return the rented array. 

So maybe we can just do this?
```fsharp
use bytes = pickleObjWithDanglingCcus inMem file g scope p x
```

Reply:
I think @0101 is correct here

Reply:
done